iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/SecretsGenerationCheck.java

```diff
@@ -227,42 +228,4 @@ private static ArgumentResolution getLastArgument(CommandDetector.Command comman
     List<ArgumentResolution> arguments = command.getResolvedArguments();
     return arguments.get(arguments.size() - 1);
   }
-
-  private static class HtpasswdDetector {
```

Comment:
Too long nested class; SQ suggested to extract it

Reply:
While I agree it's not ideal to have too long nested class, I'm not sure if it's a good idea to extract it, because to me this class behavior is really specific to this rule, checking precise values such as `htpasswd`, `-b` and `-n`.
By extracting it outside, we send the implicit message that it can be useful in other checks, which seems unlikely to me.
Also, I didn't think 36 lines to be that long :sweat_smile:
WDYT?

Reply:
On the one hand I agree that it doesn't make sense to make this logic public so to say. On the other hand, that class where it's used is already pretty long and IMO could benefit from being split.

Reply:
But yeah, I've decided to rollback this extraction after all